packages/sdk/src/wasm/VerifiablePresentationV1/audit-record.ts

```diff
@@ -101,6 +114,212 @@ export function create(
+ * context (if present).
+ *
+ * @param verificationRequest - the origin verification request for the presentation.
+ * @param presentation - the presentation for the origin verification request.
+ * @param grpc - a GRPC client for looking up the verification request anchor.
+ *
+ * @throws if any part of the context comparison fails
+ */
+async function compareContexts(
+    { context: rc, transactionRef }: VerificationRequestV1.Type,
+    { presentationContext: pc }: VerifiablePresentationV1.Type,
+    grpc: ConcordiumGRPCClient
+): Promise<void> {
+    rc.given.length === pc.given.length ||
+        bail(
+            `Mismatch in number of given context items: request context has ${rc.requested.length} while presentation context has ${pc.requested.length}`
+        );
+    rc.requested.length === pc.requested.length ||
+        bail(
+            `Mismatch in number of requested context items: request context has ${rc.requested.length} while presentation context has ${pc.requested.length}`
+        );
+
+    rc.given.every(
+        (rgc) =>
+            pc.given.some((pgc) => contextEquals(rgc, pgc)) ||
+            bail(`No matching context found for given request context ${rgc.label}`)
+    );
+    rc.requested.every(
+        (rrc) =>
+            pc.requested.some((prc) => prc.label === rrc) ||
+            bail(`No matching context found for requested request context ${rrc}`)
+    );
+
+    const blockHashContext = pc.requested.find((prc) => prc.label === 'BlockHash');
+    if (blockHashContext === undefined) return;
+
+    if (!BlockHash.instanceOf(blockHashContext.context))
+        throw new Error('Expected "BlockHash" type for "BlockHash" context');
+
+    const blockItemStatus = await grpc.getBlockItemStatus(transactionRef);
+    if (blockItemStatus.status !== TransactionStatusEnum.Finalized)
+        throw new Error('Verification request anchor not finalized');
+
+    if (!BlockHash.equals(blockItemStatus.outcome.blockHash, blockHashContext.context))
+        throw new Error('Block hash from presentation does not match block found for verification request anchor');
+}
+
+function verifyAtomicStatements<A>(a: AtomicStatementV2<A>[], b: AtomicStatementV2<A>[]): void {
+    if (a.length !== b.length) throw new Error('Mismatch in number of atomic statements for statement/claim');
+
+    const atomicError = new Error(
+        'Mismatch found when comparing atomic statements for verification/presentation request.'
+    );
+
+    a.forEach((as, i) => {
+        const bs = b[i];
+        switch (as.type) {
+            case 'AttributeInSet': {
+                if (bs.type !== 'AttributeInSet') throw atomicError;
+                if (as.set.length !== bs.set.length) throw atomicError;
+                if (!as.set.every((asv) => bs.set.includes(asv))) throw atomicError;
+                break;
+            }
+            case 'AttributeNotInSet': {
+                if (bs.type !== 'AttributeNotInSet') throw atomicError;
```

Comment:
What about the attributeTag. Should be compared as well. Otherwise one statement is about citizenship but the other statement is about nationality